src/pages/update.tsx

```diff
@@ -37,7 +38,7 @@ export default function UpdateDetail() {
     return <LoadingView />
   }
 
-  const index = publicUpdates && publicUpdates.length - Number(publicUpdates?.findIndex((item) => item.id === updateId))
+  const index = (!!publicUpdates && !!updateId && getUpdateNumber(publicUpdates, updateId)) || NaN
```

Comment:
I think this checks can be encapsulated in  `getUpdateNumber`, because it is already returning `NaN`

We could make that function to accept any input and return `NaN` if anything is not valid.

```typescript
const index = getUpdateNumber(publicUpdates, updateId))

export function getUpdateNumber(publicUpdates?: UpdateAttributes[], updateId?: string ) {
  if(!publicUpdates || !updateId) return NaN
  const updateIdx = Number(publicUpdates.findIndex((item) => item.id === updateId))
  return publicUpdates.length > 0 && updateIdx >= 0 ? publicUpdates.length - updateIdx : NaN
}
```

Reply:
Also, what happens in line 50 if index is `NaN`? 
Shouldn't we handle it the same way as `updateState.error` above?